compiler/noirc_evaluator/src/acir/tests/mod.rs

```diff
@@ -926,3 +927,208 @@ fn do_not_overflow_with_constant_constrain_neq() {
     assert_eq!(acir_functions.len(), 1);
     assert!(acir_functions[0].opcodes().is_empty());
 }
+
+// Convert the SSA input into ACIR and use ACVM to execute it
+// Returns the ACVM execution status and the value of the 'output' witness value,
+// unless the provided output is None or the ACVM fails during execution.
+fn execute_ssa(
+    ssa: Ssa,
+    initial_witness: WitnessMap<FieldElement>,
+    output: Option<&Witness>,
+) -> (ACVMStatus<FieldElement>, Option<FieldElement>) {
+    let brillig = ssa.to_brillig(&BrilligOptions::default());
+    let (acir_functions, brillig_functions, _, _) = ssa
+        .into_acir(&brillig, &BrilligOptions::default(), ExpressionWidth::default())
+        .expect("Should compile manually written SSA into ACIR");
+    assert_eq!(acir_functions.len(), 1);
+    let main = &acir_functions[0];
+    let mut acvm = ACVM::new(
+        &StubbedBlackBoxSolver(true),
+        main.opcodes(),
+        initial_witness,
+        &brillig_functions,
+        &[],
+    );
+    let status = acvm.solve();
+    if status == ACVMStatus::Solved {
+        (status, output.map(|o| acvm.witness_map()[o]))
+    } else {
+        (status, None)
+    }
+}
+
+fn get_main_src(typ: &str) -> String {
+    format!(
+        "acir(inline) fn main f0 {{
+            b0(inputs: [{typ}; 2]):
+              lhs = array_get inputs, index u32 0 -> {typ}
+              rhs = array_get inputs, index u32 1 -> {typ}
+              "
+    )
+}
+
+// Create a ssa instruction corresponding to the operator, using v1 and v2 as operands.
+// Additional information can be added to the string,
+// for instance, "range_check 8" creates 'range_check v1 to 8 bits'
+fn generate_test_instruction_from_operator(operator: &str) -> (String, bool) {
+    let ops = operator.split(" ").collect::<Vec<_>>();
+    let op = ops[0];
+    let mut output = true;
+    let src = match op {
+        "constrain" => {
+            output = false;
+            format!("constrain lhs {} rhs", ops[1])
+        }
+        "not" => format!("result = {} lhs", op),
+        "truncate" => {
+            format!("result = truncate lhs to {} bits, max_bit_size: {}", ops[1], ops[2])
+        }
+        "range_check" => {
+            output = false;
+            format!("range_check lhs to {} bits", ops[1])
+        }
+        _ => format!("result = {} lhs, rhs", op),
+    };
+
+    if output {
+        (
+            format!(
+                "
+            {src}
+        return result
+        }}"
+            ),
+            true,
+        )
+    } else {
+        (
+            format!(
+                "
+            {src}
+        return
+        }}"
+            ),
+            false,
+        )
+    }
+}
+
+// Execute a simple operation for each operators
+// The operation is executed from SSA IR using ACVM after acir-gen
+// and also directly on the SSA IR via constant folding, by hardcoding the values
```

Comment:
Looks like this comment needs to be update that it is testing against the SSA interpreter

Reply:
Done